tests/compile_tests.sh

```diff
@@ -56,9 +56,14 @@ do
         compilerFlags=$(cat "./tests/compile_tests/$category/$testName.flags")
     fi
 
+    pushd . >/dev/null
+    cd "./tests/compile_tests/$category"
+
     # -m and -q make sure that we get predictable results
-    "$QB64" $compilerFlags -m -q -x "./tests/compile_tests/$category/$testName.bas" -o "$EXE" 1>"$compileResultOutput"
+    "../../../$QB64" $compilerFlags -m -q -x "$testName.bas" -o "$EXE" 1>"../../../$compileResultOutput"
```

Comment:
Taking a closer look the `-o` flag might be a problem, I didn't notice but it's still relative to the location of qb64pe when using it. Unfortunately unlike with selecting the source file I'm not sure there's a simple non-breaking solution for this one, I'll have to think about it. We could add a new flag for it but that's a bit ugly...

Reply:
After some thinking I we can fix it in a non-breaking way, we can just make it so that the `-o` location is relative to the same location the source file was found relative too. I'll get that done in this PR and we can review it again.

My logic is that it's not a breaking change because currently we only allowed source files relative to `qb64pe`, and in that situation we'll treat the `-o` location as relative to `qb64pe`, the same as it currently works. In the new situation of a source file that's relative to the cwd/`_STARTDIR$` that just gave an error before so it has no defined behavior, making `-o` relative to the source location thus doesn't break any previously working usages. Additionally I think it's by far the expected behavior, so it makes sense to make that change.